db/src/master.rs

```diff
@@ -776,6 +777,7 @@ impl Master {
                 // If there is a value, then write it in.
                 if val.len() > 0 {
                     status = RpcStatus::StatusInternalError;
+                    let alloc = unsafe { Box::from_raw(alloc as *mut Allocator) };
```

Comment:
In this case, we can't pass the reference as the compiler complains about the lifetime of a reference without a type. So, I need to wrap the pointer in some type(Box, Rc, Arc).

Logically, it should be Arc because multiple threads are sharing this reference and wrong reference count can result in deallocation of the object. And now I am not sure why Box is working.

Please share your views about it.